lib/msgr/client.rb

```diff
@@ -140,7 +140,7 @@ def sync_publish(payload, opts)
       begin
         payload = JSON.dump(payload)
         opts[:content_type] ||= 'application/json'
-      rescue
+      rescue JSON::JSONError
```

Comment:
I hope this is the right one. Was a bit hard to find an explicit statement about this in the Ruby docs.